driver-core/src/main/com/mongodb/operation/CommandOperationHelper.java

```diff
@@ -105,47 +107,108 @@ public BsonDocument apply(final BsonDocument command) {
+                    }
+                    logRetryExecute(command.getFirstKey(), originalException);
+                    return transformer.apply(executeWrappedCommandProtocol(database, command, decoder, source,
+                            binding.getReadPreference()),
+                            source.getServerDescription().getAddress());
+                } finally {
+                    connection.release();
+                }
+            }
+        });
     }
 
-    static BsonDocument executeWrappedCommandProtocol(final ReadBinding binding, final String database, final BsonDocument command,
-                                               final Connection connection) {
-        return executeWrappedCommandProtocol(binding, database, command, connection, new IdentityTransformer<BsonDocument>());
+    static BsonDocument executeCommand(final ReadBinding binding, final boolean retryReads, final String database,
+                                       final BsonDocument command, final Connection connection) {
+        return executeCommand(binding, retryReads, database, command, connection, new IdentityTransformer<BsonDocument>());
     }
 
-    static <T> T executeWrappedCommandProtocol(final ReadBinding binding, final String database, final BsonDocument command,
-                                               final Connection connection, final CommandTransformer<BsonDocument, T> transformer) {
-        return executeWrappedCommandProtocol(binding, database, command, new BsonDocumentCodec(), connection, transformer);
+    static <T> T executeCommand(final ReadBinding binding, final boolean retryReads, final String database, final BsonDocument command,
+                                final Connection connection, final CommandTransformer<BsonDocument, T> transformer) {
+        return executeCommand(binding, retryReads, database, command, new BsonDocumentCodec(), connection, transformer);
     }
 
-    static <T> T executeWrappedCommandProtocol(final ReadBinding binding, final String database, final BsonDocument command,
-                                               final Decoder<BsonDocument> decoder, final Connection connection,
-                                               final CommandTransformer<BsonDocument, T> transformer) {
-        return executeWrappedCommandProtocol(database, command, decoder, connection, binding.getReadPreference(), transformer,
-                binding.getSessionContext());
+    static <T> T executeCommand(final ReadBinding binding, final boolean retryReads, final String database, final BsonDocument command,
+                                final Decoder<BsonDocument> decoder, final Connection connection,
+                                final CommandTransformer<BsonDocument, T> transformer) {
+        MongoException exception;
+
+        try {
+            return executeWrappedCommandProtocol(database, command, decoder, connection, binding.getReadPreference(), transformer,
```

Comment:
Nit: rename all the executeWrappedCommandProtocol methods to just executeCommand

Reply:
Done.